toolkit/tools/pkg/imagemodifierlib/modifierutils.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagemodifierlib
+
+import (
+	"strconv"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/installutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/userutils"
+)
+
+func doModifications(baseConfigPath string, config *imagecustomizerapi.Config) error {
+	err := addOrUpdateUsers(config.SystemConfig.Users, baseConfigPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addOrUpdateUsers(users []imagecustomizerapi.User, baseConfigPath string) error {
+	for _, user := range users {
+		err := addOrUpdateUser(user, baseConfigPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addOrUpdateUser(user imagecustomizerapi.User, baseConfigPath string) error {
```

Comment:
For basic functions, splitting the functions into a chroot variant and in-OS variant is fairly straightforward. But once we start dealing with these higher-level functions, it starts to become a lot messier.

It is possible that my original idea of having a chroot interface and a dummy implementation for in-OS scenarios, might be a little cleaner.

Reply:
make sense

Reply:
I have a commit of this change, could you take a look at https://github.com/microsoft/CBL-Mariner/compare/main...elaine/add-initial-im2?diff=split and let me know if this is the right path?

Reply:
Yeah, that’s looks far cleaner to me.

(By the way, the RootDir() should probably be “/“.)